locator/snitch_base.hh

```diff
@@ -209,6 +204,10 @@ struct snitch_ptr : public peering_sharded_service<snitch_ptr> {
         }
     }
 
+    i_endpoint_snitch& get() {
```

Comment:
Please, make it smth like
```c++
template <typename T>
T& get_as<T>() {
    return dynamic_cast<T&>(*_ptr);
}
```

there are more places over the snitch where snitch implementations need to call each-other cross-shard and have no choice by to use i_endpoint_snitch's virtual functions